In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

df = pd.read_csv('ForeignGifts_edu.csv',low_memory=False)
df.head()

,ID,OPEID,Institution Name,City,State,Foreign Gift Received Date,Foreign Gift Amount,Gift Type,Country of Giftor,Giftor Name
0,1,102000,Jacksonville State University,Jacksonville,AL,43738,250000,Monetary Gift,CHINA,NaN
1,2,104700,Troy University,Troy,AL,43592,463657,Contract,CHINA,Confucius Institute Headquarters
2,3,105100,University of Alabama,Tuscaloosa,AL,43466,3649107,Contract,ENGLAND,Springer Nature Customer Service Ce
3,4,105100,University of Alabama,Tuscaloosa,AL,43472,1000,Contract,SAUDI ARABIA,Saudi Arabia Education Mission
4,5,105100,University of Alabama,Tuscaloosa,AL,43479,49476,Contract,SAUDI ARABIA,Saudi Arabia Education Mission


In [4]:
#UVA stats 
uva_only = df[df["Institution Name"]=="University of Virginia"]

#total count: 98
len(uva_only)

#average gift size: $226420
uva_mean = uva_only["Foreign Gift Amount"].mean()

#total: 22189238
uva_total = uva_only["Foreign Gift Amount"].sum()

#contract: 64 monetary gift: 34 
print(uva_only["Gift Type"].value_counts())

#top donors: 
uva_giftor_total = uva_only.groupby("Giftor Name")["Foreign Gift Amount"].sum()
uva_giftor_total.sort_values(ascending=False)
#Top 5: AstraZeneca, Intrnl Centre Diarrheal Disease, Haydom Lutheran Hospital d/b/a Hayd, Haydom Luther Hospital, Harvest Group Holding
print(uva_total)
print(uva_giftor_total.sort_values(ascending=False))


Gift Type
Contract         64
Monetary Gift    34
Name: count, dtype: int64
22189238
Giftor Name
AstraZeneca                            3208048
Intrnl Centre Diarrheal Disease        1382896
Haydom Lutheran Hospital d/b/a Hayd    1100553
Haydom Luther Hospital                 1035339
Harvest Group Holding                  1000000
Lawrence D/ Karin S Howell             1000000
Lawrence D & Karin S Howell            1000000
Danisco Sweeteners Oy                   990331
Intnl Centre for Diarrhoeal Disease     919857
International Centre for Diarrhoeal     919518
Hemera Foundation                       900000
Rainbow Foundation                      805000
Charities Aid Foundation                727928
Intl. Cntr for Diarrihoeal Disease      696317
AutoLiv Development AB                  685217
Haydom Lutheran Hospital                640555
Hyundai Motor Company                   568092
Harvest Group Holding (PTC Limited)     500000
Kirkhouse Trust                         366460
Charles Ai

In [5]:
#Comapring UVA to peer institutions (colleges in Virginia)
all_VA_rows=df[df["State"]=="VA"]
all_VA_rows["Institution Name"].unique()
#College of William & Mary, George Mason University, Virginia Commonwealth University, Virginia Polytechnic Institute & State University

va_total = all_VA_rows.groupby("Institution Name")["Foreign Gift Amount"].sum()
va_total.mean()
print(va_total.mean())
#38,392,997

#Comparing to UNC and University of Michigan
df[df["Institution Name"]=="University of North Carolina - Chapel Hill"]
df[df["Institution Name"]=="University of Michigan - Ann Arbor"]

peer_institutions = ["College of William & Mary", "George Mason University", "Virginia Commonwealth University", "Virginia Polytechnic Institute & State University", "University of North Carolina - Chapel Hill","University of Michigan - Ann Arbor" ]

def get_stats(college_name):
    place_only = df[df["Institution Name"]==college_name]
    count = len(place_only)
    place_mean = place_only["Foreign Gift Amount"].mean()
    place_total = place_only["Foreign Gift Amount"].sum()
    gift_count = place_only["Gift Type"].value_counts()
    place_giftor_total = place_only.groupby("Giftor Name")["Foreign Gift Amount"].sum()
    sorted_place_val = place_giftor_total.sort_values(ascending=False)
    top5_place = sorted_place_val.head(5)
    print(college_name)
    print(f"Number of rows: {count} //")
    print(f"Mean: {place_mean} //")
    print(f"Total: {place_total} //")
    print(f"Gift Types: {gift_count} //")
    print(f"Top 5 donors: {top5_place} //")

for college in peer_institutions:
    get_stats(college)





38392997.0
College of William & Mary
Number of rows: 17 //
Mean: 199088.11764705883 //
Total: 3384498 //
Gift Types: Gift Type
Monetary Gift    17
Name: count, dtype: int64 //
Top 5 donors: Giftor Name
Royal Embassy of Saudi Arabia/SACM     2262530
Beijing Normal University               510861
Hanban                                  309453
Multiple Individuals                    263618
Hanban North American Education Inc      38036
Name: Foreign Gift Amount, dtype: int64 //
George Mason University
Number of rows: 38 //
Mean: 1721989.5789473683 //
Total: 65435604 //
Gift Types: Gift Type
Contract    38
Name: count, dtype: int64 //
Top 5 donors: Giftor Name
Royal Embassy of Saudi Arabia          54183796
Embassy of the United Arab Emirates     4406188
Embassy of the State of Kuwait          1388173
Embassy of The UAE - Police Attache     1128033
Emg of the State of Qatar Cultural       813266
Name: Foreign Gift Amount, dtype: int64 //
Virginia Commonwealth University
Number of rows: 115

Results: 
Institution Name
College of William & Mary                             3384498
George Mason University                              65435604
University of Virginia                               22189238
Virginia Commonwealth University                     89163583
Virginia Polytechnic Institute & State University    11792062
Average: 38,392,997

In [52]:
#Comparing UVA to overall Market (using numbers from inital_analysis.ipynb)

each_country_total = df.groupby("Country of Giftor")["Foreign Gift Amount"].sum()
each_country_total.sort_values(ascending=False)
each_country_total_top5 = each_country_total.sort_values(ascending=False).head(5)

each_institution_total = df.groupby("Institution Name")["Foreign Gift Amount"].sum()
each_institution_avg = each_institution_total.mean()
each_institution_median = each_institution_total.median()

each_giftor_total = df.groupby("Giftor Name")["Foreign Gift Amount"].sum()
each_giftor_total.sort_values(ascending=False)
each_gifter_total_top5 = each_giftor_total.sort_values(ascending=False).head(5)

print(each_country_total_top5)
print(each_institution_avg)
print(each_institution_median)
print(each_gifter_total_top5)


Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
Name: Foreign Gift Amount, dtype: int64
52202878.974842764
6486791.5
Giftor Name
Qatar Foundation                       1166503744
Qatar Foundation/Qatar National Res     796197000
Qatar Foundation for Education          373945215
Anonymous                               338793629
Saudi Arabian Cultural Mission          275221475
Name: Foreign Gift Amount, dtype: int64


In [46]:
print(uva_only[uva_only["Giftor Name"]=="Qatar Foundation"])
print(uva_only[uva_only["Giftor Name"]=="Royal Embassy of Saudi Arabia"])
print(uva_only[uva_only["Country of Giftor"]=="QATAR"])
print(uva_only[uva_only["Country of Giftor"]=="CHINA"])
print(uva_only[uva_only["Country of Giftor"]=="SAUDI ARABIA"])

Empty DataFrame
Columns: [ID, OPEID, Institution Name, City, State, Foreign Gift Received Date, Foreign Gift Amount, Gift Type, Country of Giftor, Giftor Name]
Index: []
Empty DataFrame
Columns: [ID, OPEID, Institution Name, City, State, Foreign Gift Received Date, Foreign Gift Amount, Gift Type, Country of Giftor, Giftor Name]
Index: []
Empty DataFrame
Columns: [ID, OPEID, Institution Name, City, State, Foreign Gift Received Date, Foreign Gift Amount, Gift Type, Country of Giftor, Giftor Name]
Index: []
Empty DataFrame
Columns: [ID, OPEID, Institution Name, City, State, Foreign Gift Received Date, Foreign Gift Amount, Gift Type, Country of Giftor, Giftor Name]
Index: []
Empty DataFrame
Columns: [ID, OPEID, Institution Name, City, State, Foreign Gift Received Date, Foreign Gift Amount, Gift Type, Country of Giftor, Giftor Name]
Index: []


Result: 
UVA's total foreign gift amount is around $22 million. Comapred to peer institutions, UVA is below the average. The average total a college in Virginia recieved was around $38 million. When comapred to other large public universities (UNC and University of Michigan - Ann Arbor) similar to UVA, UVA's total amount also falls below the two institutions. UNC's total is around $41 million and Michigan's total was around $287 million. However, when calculating the median of total amount recieved across all listed institutions, UVA performs significantly better (avg:$6.5 vs UVA: $22 million). Aross colleges, the avgerage is around $52 million while the median is $6.5 million. 

The first way UVA can increase its funding is by signing more contracts. VCU engaged in 115 contract, UNC had 267, and Michigan was involved in 643 contracts. UVA only had 64 contracts and 34 monetary gifts. 
The second way is to connect with countries that are known to give large sums of money to American instiutions. UVA's top 5 donor does not include any giftors from Qatar or China or Saudi Arabia. All three of the countries are in the top five of largest sums of money given to colleges. VCU's largest donor is the Qatar Foundation and GMU's largest donor is the Royal Embassy of Saudi Arabia. UVA may be able to raise its foreign gift amount by focusing their time on building relationships with those identified giftors. 